website/docs/reference/opspec/index.md

```diff
@@ -101,5 +34,55 @@ Reference:
 - [socket](types/socket.md)
 - [string](types/string.md)
 
-## Scoping
-Variables in opspec are scoped to each operation. [Parameters](../op/parameter/index.md) allow passing values into or out of this scope.
\ No newline at end of file
+### Scoping
+
+A variable is created when defined as an [`input`](op.yml#inputs) or [`output`](op.yml#inputs), bound implicitly to an `output` when making an [op call](op.yml/call/op), or declared with [loop variables](./op.yml/loop-variable). A variable name is in scope from creation through the lifetime of it's `op.yml` file. Variable scope does not extend into [op calls](./op.yml/call/op).
+
+Variables can be referenced using the syntax `$(name)`.
+
+#### Scoping example
+
+```yml
+name: foobar
+inputs:
+  variable1:
+    string: {}
+  variable2:
+    string:
+      default: "hello world"
+outputs:
+  variable3:
+    string:
+      default: "goodbye!"
+```
+
+Two variables, `variable1` and `variable2` are created from inputs. The CLI will prompt the user to provide a value for `variable1`, since it doesn't have a default, and `variable2` will contain the string value `"hello world"` if the user didn't override the value with a command line argument.
+
+One variable, `variable3` is created from the single output, which contains the string value `"goodbye!"`.
+
+```yml
+run:
+  serial:
+    - op:
+        ref: ../other-op
+        inputs:
+          input1: $(variable1)
+        outputs:
+          output1: $(variable2)
+          output2: $(variable4)
+```
+
+The variable `variable1`'s value is passed into `../other-op` where it will be stored in the variable `input1`. `../other-op` will not have access to the `variable1` or `variable2` names. After the op call completes the variable `variable2` will contain the value from `output1` that was produced by `../other-op`. Additionally, the variable `variable3` will be implicitly created from the output binding and will have the value from `output2` stored in it.
```

Comment:
```suggestion
The variable `variable1`'s value is passed into `../other-op` where it will be stored in the variable `input1`. `../other-op` will not have access to the `variable1` name. After the op call completes the variable `variable2` will contain the value from `output1` that was produced by `../other-op`. Additionally, the variable `variable4` will be implicitly created from the output binding and will have the value from `output2` stored in it.
```

Reply:
After reading this section I still don't think I have a grasp of when/which variables are implicitly bound.

Reply:
Me neither! After my initial pass at this, I discovered that variables can be implicitly bound in loop vars.

This is a start at an explanation and will give someone a reference point to ask questions about and make improvements. I think this is a good place case for a more in depth "training" page.

Reply:
Anywhere that you would write `blah : $(blah)`, you can write `blah:`. That's when/which : ). The whole point behind implicit vars was literally to save you having to repeat yourself; that's it.